### DISCUSSION TITLE
Resume publishing locally from CI failure

### DISCUSSION THREAD

**Discussion by yschimke** [jreleaser/jreleaser#627] [Ideas]
Reading docs

> JReleaser follows a series of steps to create and upload a release. You may invoke all steps in one go (the preferred way) or each step individually and separate from the others. Steps are executed sequentially; if one of them fails then the build fails. You may decide to resume from the start or from the last failed step.

Got me wondering if I'm missing something. Is this "resume" a manual step to run the following steps?

Either way, it got me thinking about my experience of running these steps locally and in CI and trying to fix them.  I probably should have used dryrun.  But I got delayed by the missing local artifacts when trying to build locally, and in one case downloaded it from github release page to continue without waiting for a slow native-image run locally.  Could the publish steps use existing published artifacts when resuming?

**Comment by aalmiray** [comment]
Yes, that's the goal. If you're building locally and one of the steps fail then you may resume a `full-release` by invoking individual steps.

Note that in the case of platform specific artifacts you can still try a release (with or without `dryrun`) if you filter artifacts per platform. This is why `--select-current-platform` and --select-platform` options are available.

Say you're building GraalVM Native Images for all platforms and your current local environment is Linux. If you only have a Linux archive then invoking `config` or any other command that requires all artifacts to be available will fail. But if you add `--select-current-platform` you'll be able to execute the command.

**Reply by yschimke** [reply]
Yeah, I was actually hoping it could use the published artifacts from CI, and let me continue locally.  But understand if that's too much work for little pay off.

**Reply by aalmiray** [reply]
The assumption is artifacts are available from CI at a well defined path. But which CI? We support more than 14 different options although granted, GitHub Actions is the most popular one at the moment.

You'd have to configure:
- the name of the workflow
- the directory(ies) that contain artifacts
- artifact paths
- potentially the commit hash in case of needed artifacts from a previous execution that's not the latest one

It can get messy pretty quickly.

**Reply by yschimke** [reply]
I meant from the https://jreleaser.org/guide/latest/configuration/release/github.html step, which publishes to well known locations in

https://github.com/yschimke/okurl/releases/tag/v3.17.0

**Reply by aalmiray** [reply]
If the build failed you have to figure out from where. 
- If it happened before the `checksum` step some artifacts may not be available on CI. 
- If it happened before `release` then artifacts are not available as release assets as there's no release.
- if it happened after `release` then artifacts are available as release assets.

**Reply by yschimke** [reply]
I'll drop this, as it seems out of scope and way past the cost/benefit point.  I think I read too much smart into the "resume" feature, as if jreleaser would check and understand which steps had completed based on evidence, like a github release artifacts.

But I understand it's not in scope.
